src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -530,41 +551,49 @@ public BitbucketTeam getTeam() throws IOException, InterruptedException {
     @Override
     public List<BitbucketCloudRepository> getRepositories(@CheckForNull UserRoleInRepository role)
             throws InterruptedException, IOException {
-        Integer pageNumber = 1;
-        UriTemplate template = UriTemplate.fromTemplate(V2_API_BASE_URL + "{/owner}{?role,page,pagelen}")
+        final UriTemplate template = UriTemplate.fromTemplate(V2_API_BASE_URL + "{/owner}{?role,page,pagelen}")
                 .set("owner", owner)
-                .set("page", pageNumber)
                 .set("pagelen", 50);
         if (role != null && getLogin() != null) {
             template.set("role", role.getId());
         }
-        String url = template.expand();
-        List<BitbucketCloudRepository> repositories = new ArrayList<BitbucketCloudRepository>();
-        String response = getRequest(url);
-        PaginatedBitbucketRepository page;
         try {
-            page = JsonParser.toJava(response, PaginatedBitbucketRepository.class);
-            repositories.addAll(page.getValues());
-        } catch (IOException e) {
-            throw new IOException("I/O error when parsing response from URL: " + url, e);
-        }
-        while (page.getNext() != null) {
-                pageNumber++;
-                response = getRequest(url = template.set("page", pageNumber).expand());
-            try {
-                page = JsonParser.toJava(response, PaginatedBitbucketRepository.class);
-                repositories.addAll(page.getValues());
-            } catch (IOException e) {
-                throw new IOException("I/O error when parsing response from URL: " + url, e);
-            }
+            return cachedRepositories.get(template, new Callable<List<BitbucketCloudRepository>>() {
+                @Override
+                public List<BitbucketCloudRepository> call() throws Exception {
+                    List<BitbucketCloudRepository> repositories = new ArrayList<BitbucketCloudRepository>();
+                    Integer pageNumber = 1;
+                    String url = template.set("page", pageNumber).expand();
+                    String response = getRequest(url);
+                    PaginatedBitbucketRepository page;
+                    try {
```

Comment:
I think a `do...while` loop would be better than the duplicate code... I wouldn't mention except we are optimizing the code a little with this PR anyway

Reply:
Agree, will change

Reply:
done